File: backend/src/tact/schemas/project.py
from datetime import datetime

from pydantic import BaseModel


class ProjectCreate(BaseModel):
    id: str
    name: str


class ProjectUpdate(BaseModel):
    name: str | None = None
    active: bool | None = None


class ProjectResponse(BaseModel):
    id: str
    name: str
    active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: tui/model/project.go
package model

type Project struct {
	ID          string `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
	Active      bool   `json:"active"`
	CreatedAt   Time   `json:"created_at"`
	UpdatedAt   Time   `json:"updated_at"`
}
